src/addr.rs

```diff
@@ -66,27 +66,29 @@ impl VirtAddr {
     ///
     /// ## Panics
     ///
-    /// This function panics if the bits in the range 48 to 64 contain data (i.e. are not null and no sign extension).
-    #[inline]
-    pub fn new(addr: u64) -> VirtAddr {
-        Self::try_new(addr).expect(
-            "address passed to VirtAddr::new must not contain any data \
-             in bits 48 to 64",
-        )
+    /// This function panics if the bits in the range 48 to 64 are invalid
+    /// (i.e. are not a proper sign extension of bit 47).
+    #[inline]
+    pub const fn new(addr: u64) -> VirtAddr {
+        // TODO: Replace with .ok().expect(msg) when that works on stable.
+        match Self::try_new(addr) {
+            Ok(v) => v,
+            Err(_) => panic!("virtual address must be sign extended in bits 48 to 64"),
+        }
     }
 
     /// Tries to create a new canonical virtual address.
     ///
-    /// This function tries to performs sign
-    /// extension of bit 47 to make the address canonical. It succeeds if bits 48 to 64 are
-    /// either a correct sign extension (i.e. copies of bit 47) or all null. Else, an error
-    /// is returned.
-    #[inline]
-    pub fn try_new(addr: u64) -> Result<VirtAddr, VirtAddrNotValid> {
-        match addr.get_bits(47..64) {
-            0 | 0x1ffff => Ok(VirtAddr(addr)),     // address is canonical
-            1 => Ok(VirtAddr::new_truncate(addr)), // address needs sign extension
-            _ => Err(VirtAddrNotValid(addr)),
+    /// This function tries to performs sign extension of bit 47 to make the
+    /// address canonical. It succeeds if bits 48 to 64 are a correct sign
+    /// extension (i.e. copies of bit 47). Else, an error is returned.
```

Comment:
These docs are a bit confusing. The function only does a sign extension to check if the address is canonical, so I wouldn't mention that implementation detail in the docs. Maybe something like this:

```suggestion
    /// This function checks wether the given address is canonical
    /// and returns an error otherwise. An address is canoncial
    /// if bits 48 to 64 are a correct sign
    /// extension (i.e. copies of bit 47).
```